src/Model/IndexInfo.php

```diff
@@ -110,6 +200,29 @@ public function getVersion()
         return (integer) $this->info['v'];
     }
 
+    /**
+     * Return the weights document for text indexes.
+     *
+     * @return document
+     */
+    public function getWeights()
+    {
+        if (empty($this->info['weights'])) {
+            return self::DEFAULT_WEIGHT;
```

Comment:
``self::DEFAULT_WEIGHT`` is not declared as a ``document`` (array) in the class definition, but as a simple number ``1``. I think it should return ``null`` if ``weights`` is empty, but happy to hear with @jmikola has to say on this too.

Reply:
The next comment shows how ``weights`` looks like, when a text index exists: